codes/c/chapter_tree/array_binary_tree.c

```diff
@@ -6,29 +6,31 @@
 
 #include "../utils/common.h"
 
-/* 数组表示下的二叉树类 */
+/* 数组表示下的二叉树结构 */
 typedef struct {
-    vector *tree;
+    int *tree;
+    int size;
 } ArrayBinaryTree;
 
-/* 构造函数 */
-ArrayBinaryTree *newArrayBinaryTree(vector *arr) {
-    ArrayBinaryTree *newABT = malloc(sizeof(ArrayBinaryTree));
-    newABT->tree = arr;
-    return newABT;
+/* 构造方法 */
+ArrayBinaryTree *createArrayBinaryTree(int *arr, int size) {
+    ArrayBinaryTree *abt = (ArrayBinaryTree *)malloc(sizeof(ArrayBinaryTree));
+    abt->tree = arr;
```

Comment:
It is recommended to reallocate a section of space, copy the external data to the internal, and release it after subsequent use.
```c
abt->tree = malloc (sizeof(int) * size);
memcpy(abt->tree, arr, size);
```

Reply:
The second line should be:

```c
memcpy(abt->tree, arr, sizeof(int) * size);
```